main/src/ca/uwaterloo/flix/language/phase/Parser2.scala

```diff
@@ -738,6 +746,49 @@ object Parser2 {
     close(openBefore(first), TreeKind.QName)
   }
 
+  /**
+    * Returns true if the next token is (close to) a sequence of dot-separated `kind` tokens.
+    *
+    * @param tail if any kind found is in `tail` then no further dot separated tokens will be consume
+    */
+  private def atNameAllowQualified(kinds: Set[TokenKind], tail: Set[TokenKind] = Set(TokenKind.NameLowerCase))(implicit s: State): Boolean = {
```

Comment:
How will we be able to maintain this?

Reply:
yeah its not great, but the parser has no infrastructure for this

Reply:
Can you open a ticket with some notes on the problem?

Reply:
Perhaps we need a separate ticket with strange parsing things.

Reply:
Perhaps the name should be `atNameAllowQualifiedUseOnlyForRunWith`?

Reply:
maybe just `lookaheadRunWith` since its also not exactly just looking for a qualified name

Reply:
Yeah, even better.